internal/pkg/deploy/deploy.go

```diff
@@ -16,3 +36,135 @@ type ResourceEvent struct {
 	Status       string
 	StatusReason string
 }
+
+type resourceGetter interface {
+	GetResourcesByTags(resourceType string, tags map[string]string) ([]string, error)
+}
+
+type configStoreClient interface {
+	GetEnvironment(appName string, environmentName string) (*config.Environment, error)
+	ListEnvironments(appName string) ([]*config.Environment, error)
+	GetService(appName, svcName string) (*config.Service, error)
+}
+
+// Store is in charge of fetching the service deployment information.
+type Store struct {
+	rgClient     resourceGetter
+	configStore  configStoreClient
+	initRgClient func(*Store, string, string) error
+}
+
+// NewStore returns a new store.
+func NewStore() (*Store, error) {
+	store, err := config.NewStore()
+	if err != nil {
+		return nil, fmt.Errorf("connect to config store: %w", err)
+	}
+	return &Store{
+		configStore: store,
+		initRgClient: func(s *Store, appName, envName string) error {
+			env, err := s.configStore.GetEnvironment(appName, envName)
+			if err != nil {
+				return fmt.Errorf("get environment %s, %w", envName, err)
+			}
+			p := session.NewProvider()
+			sess, err := p.FromRole(env.ManagerRoleARN, env.Region)
+			if err != nil {
+				return fmt.Errorf("create new session from env role: %w", err)
+			}
+			s.rgClient = rg.New(sess)
+			return nil
+		},
+	}, nil
```

Comment:
We can get rid of the extra-parameter by declaring  `initRgClient` as a [closure](https://tour.golang.org/moretypes/25).
```go
	s := &Store{
		configStore: store,
	}
	s.initRgClient = func(appName, envName string) error {
		env, err := s.configStore.GetEnvironment(appName, envName)
		if err != nil {
			return fmt.Errorf("get environment config %s: %w", envName, err)
		}
		sess, err := session.NewProvider().FromRole(env.ManagerRoleARN, env.Region)
		if err != nil {
			return fmt.Errorf("create new session from env role: %w", err)
		}
		s.rgClient = rg.New(sess)
		return nil
	}
	return s
```

Reply:
nice handling!